lib/node-utils/index.ts

```diff
@@ -525,21 +525,27 @@ const matcherNamesHandlePromise = [
 	'rejects',
 	'toResolve',
 	'toReject',
+	'toBeRejected',
+	'toBeRejectedWith',
+	'toBeRejectedWithError',
+	'toBePending',
+	'toBeResolved',
+	'toBeResolvedTo',
 ];
 
 /**
- * Determines whether a node belongs to an async assertion
- * fulfilled by `resolves` or `rejects` properties or
- *  by `toResolve` or `toReject` jest-extended matchers
- *
+ * Determines whether a node belongs to an async assertion that is fulfilled by:
+ * - `resolves` or `rejects` properties
+ * - `toResolve` or `toReject` jest-extended matchers
+ * - jasmine async matchers
  */
-export function hasClosestExpectResolvesRejects(node: TSESTree.Node): boolean {
+export function hasClosestExpectHandlesPromise(node: TSESTree.Node): boolean {
 	if (
 		isCallExpression(node) &&
 		ASTUtils.isIdentifier(node.callee) &&
 		node.parent &&
 		isMemberExpression(node.parent) &&
-		node.callee.name === 'expect'
+		['expect', 'expectAsync'].includes(node.callee.name)
```

Comment:
I'm not sure if we should check that `expect` is combined with the right subset handlers (`resolves`, `rejects`, etc), and `expectAsync` is combined with its subset handlers (`toBeRejectedWith`, `toBeResolved`, etc).

I think it's fine as it is, so we don't need to check if the combination between expect and matcher exists, as the code won't work if doesn't.

Reply:
i had the same thought and came to the same conclusion. there will be plenty of other signals elsewhere if an incompatible combination is used - TS, runtime errors, etc.